applications/StructuralMechanicsApplication/custom_constitutive/elastic_isotropic_3d.h

```diff
@@ -360,6 +360,21 @@ class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) ElasticIsotropic3D
         const ProcessInfo& rCurrentProcessInfo
         ) override;
 
+
+    /**
+     * @brief This method retrieves a material property, in thermal CL
+     * this method calculate the value of the property according to TEMPERATURE
+     * @param rVariable The property of the material to be retrieved
+     * @param rValues The constitutive parameters
+     */
+    virtual double GetMaterialProperty(
+        const Variable<double>& rVariable,
+        ConstitutiveLaw::Parameters& rValues
+        )
+    {
+        return rValues.GetMaterialProperties()[rVariable];
```

Comment:
sorry! there is no if, only the virtualization of this method!! @philbucher

Reply:
That's why we can do it with an if constexpr in C++17... (or we can even do it in C++11 with some https://en.cppreference.com/w/cpp/types/conditional and some magic lambda, do you want to try it?)

Reply:
@AlejandroCornejo ah ok thats also what I had in mind

@loumalouomega I don't see what exactly you want to do
if you try can you please make a derived branch? I am already losing the overview here :sweat_smile:

Reply:
Yes, I've talked with @loumalouomega about this, we will make improvements in another branch

Reply:
> That's why we can do it with an if constexpr in C++17... (or we can even do it in C++11 with some [en.cppreference.com/w/cpp/types/conditional](https://en.cppreference.com/w/cpp/types/conditional) and some magic lambda, do you want to try it?)

Lambda and std::function are similar in nature, but in order to implement requires a little extra of work, so I would do it in a independent PR (or I can dedicate some minutes to this)